### PR TITLE
Fix Math.hypot polyfill to call with zero/one params

### PR DESCRIPTION
**Pull request by teppeis**
[A compat-table test](https://kangax.github.io/compat-table/es6/#test-Math_methods_Math.hypot) fails due to this issue.
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/hypot

### CODE REVIEW

**Comment by MatrixFrog** [comment]
lgtm

**Comment by MatrixFrog** [comment]
Internal review was more involved than I expected. Going to back-burner this for now.

**Comment by concavelenz** [comment]
As I mentioned in the internal review, I don't think we want to change the type signature but it should be compatible with the actual implementation and I suggest we use V8's.  Which looks like:

```js
function MathHypot(x, y) { // Function length is 2.
    var length = arguments.length;
    var args = new Array(length);
    var max = 0;
    for (var i = 0; i < length; i++) {
      var n = arguments[i];
      if (typeof n !== "number") n = Number(n);
      if (n === Infinity || n === -Infinity) return Infinity;
      n = Math.abs(n);
      if (n > max) max = n;
      args[i] = n;
    }

    // Kahan summation to avoid rounding errors.
    // Normalize the numbers to the largest one to avoid overflow.
    if (max === 0) max = 1;
    var sum = 0;
    var compensation = 0;
    for (var i = 0; i < length; i++) {
      var n = args[i] / max;
      var summand = n * n - compensation;
      var preliminary = sum + summand;
      compensation = (preliminary - sum) - summand;
      sum = preliminary;
    }
    return Math.sqrt(sum) * max;
  };
```

**Comment by concavelenz** [comment]
I changed my mind, this function isn't used enough for it to valuable to have a tighter signature.

**Comment by brad4d** [comment]
@MatrixFrog are you proceeding with this?
@concavelenz is there any reason not to proceed. I think you cancelled out your previous objection.

**Comment by teppeis** [comment]
I do not know if I should fix this PR ...
I will fix If you need.

**Comment by MatrixFrog** [comment]
Probably not.

**Comment by teppeis** [comment]
@MatrixFrog @brad4d @concavelenz any updates?

**Comment by brad4d** [comment]
Created internal Google issue b/121320889

@concavelenz could you determine whether we should accept or reject this PR?

File: externs/es6.js [hunk 1]
```diff
@@ -147,7 +147,6 @@ Math.sign = function(value) {};
 Math.cbrt = function(value) {};
 
 /**
- * @param {number} value1
  * @param {...number} var_args
```
Comment (brad4d) [inline externs/es6.js]: You need to delete the `value1` argument from the parameter list, too.

**Comment by brad4d** [comment]
@concavelenz has confirmed to me that he is OK with accepting this PR.
There's one mistake I see in it for which I've added a comment.

File: externs/es6.js (hunk 1 above)
Comment (teppeis) [reply externs/es6.js]: @brad4d Nice catch! I've fixed.

**Comment by brad4d** [comment]
I've imported this CL and sent it for internal testing and review.

**Comment by brad4d** [comment]
I submitted the internal version of this change a few minutes ago.
Our next daily push to GitHub should include it and close this PR.

**Comment by teppeis** [comment]
@brad4d thanks!

**Pull request closed without merging**